File: 0187-repeated-dna-sequences/0187-repeated-dna-sequences.java
public class Solution {
    public List<String> findRepeatedDnaSequences(String s) {
        Set<String> seen = new HashSet<>();
        Set<String> res = new HashSet<>();

        for (int l = 0; l < s.length() - 9; l++) {
            String cur = s.substring(l, l + 10);
            if (seen.contains(cur)) {
                res.add(cur);
            }
            seen.add(cur);
        }
        return new ArrayList<>(res);
    }
}

File: 0187-repeated-dna-sequences/0187-repeated-dna-sequences.ts
function findRepeatedDnaSequences(s: string): string[] {
    const seen: Set<string> = new Set();
    const res: Set<string> = new Set();

    for (let l = 0; l <= s.length - 10; l++) {
        const cur: string = s.substring(l, l + 10);

        if (seen.has(cur)) {
            res.add(cur);
        }

        seen.add(cur);
    }

    return Array.from(res);
}